### DISCUSSION TITLE
Feature request: Make the use of Umbraco Cloud ID configurable

### DISCUSSION THREAD

**Discussion by PGGMTeam5** [umbraco/Umbraco.Cloud.Issues#818] [Ideas]
### Issue description

In our company we have the requirement to use AD FS or Azure AD as login providers for the Umbraco backoffice login.
After several and long chats with Umbraco support, I finally was told that it is not possible to combine Umbraco Cloud ID with a custom login provider (like AD FS or Azure AD).

But, to be able to use a custom login provider, the Umbraco Cloud ID should be disabled.

The only way to to this at this moment, is by deleting the "Umbraco.Cloud.Identity.Cms.dll".
See also: [Umbraco forum](https://our.umbraco.com/forum/using-umbraco-and-getting-started/105031-umbraco-cloud-single-sign-on-vs-ad-azure-login)

But:
1. This is a rather crude method to disable the Umbraco Cloud ID.
2. In this way it is not possible to make a per environment decision whether to use Umbraco Cloud ID or not. 
For instance only use Umbraco Cloud ID on the local development environment (for which AD FS or Azure AD is not possible or needed), or only use the custom login provider on the Live environment.

So instead of having to delete a dll from the bin folder, could a boolean appsetting like "Umbraco.Cloud.Identity.UseUmbracoCloudID" be introduced?!

**Comment by sajumb** [comment]
Thanks for the feature request. We have created a product backlog item for this. I will let you know if it is decided that we will implement this feature.

**Comment by ronaldbarendse** [comment]
Instead of deleting the DLL (which will get restored on a new deployment), you can also leave it installed and [disable the composer](https://our.umbraco.com/documentation/implementation/composing/#disablecomposer--disable) instead using:

```c#
// For v3 (targeting Umbraco 8):
[assembly:DisableComposer(typeof(Umbraco.Cloud.Identity.Cms.UmbracoIdComposer))]

// For v4 (targeting Umbraco 9):
[assembly:DisableComposer(typeof(Umbraco.Cloud.Identity.Cms.V9.UmbracoIdComposer))]

// For v10 (targeting Umbraco 10):
[assembly:DisableComposer(typeof(Umbraco.Cloud.Identity.Cms.V10.UmbracoIdComposer))]
```

**Comment by LennardF1989** [comment]
Tried this for v14 @ronaldbarendse:
```
[assembly:DisableComposer(typeof(Umbraco.Cloud.Identity.Cms.V14.UmbracoIdComposer))]
```

It seems the "Umbraco Id" button is still shown on `/umbraco`, but clicking it ends in an exception. Simply leaving the whole `<PackageReference Include="Umbraco.Cloud.Identity.Cms" Version="14.0.1" />` reference out of the .csproj reverts the login screen back to the normal username/password one.

Is there something else that needs to be disabled to get rid of the button gracefully (as in: without resorting to having to remove the DLL). And/or: is there a better way of disabling it by now?

**Comment by jemayn** [comment]
Hey everyone!

I am going through some of the older issues and feature requests on this tracker. I think the original issue here is resolved with the External Login Providers feature that has since been added to Umbraco Cloud: https://docs.umbraco.com/umbraco-cloud/begin-your-cloud-journey/project-features/external-login-providers

If not please create a new dicussion and let us know what parts you would want! 🙂
